packages/action-group/test/action-group.test.ts

```diff
@@ -659,6 +699,54 @@ describe('ActionGroup', () => {
         expect(secondButton.selected, 'second button selected').to.be.true;
     });
 
+    it('does not allow interaction with child content to interupt the selection mechanism', async () => {
+        const el = await singleSelectedActionGroup([]);
+        await elementUpdated(el);
+        expect(el.selected.length).to.equal(0);
+
+        const firstButton = el.querySelector('.first') as ActionButton;
+        const secondButton = el.querySelector('.second') as ActionButton;
+        const icon = secondButton.querySelector('[slot=icon]') as HTMLElement;
+        expect(firstButton.selected, 'first button selected').to.be.false;
+        expect(secondButton.selected, 'second button not selected').to.be.false;
+
+        secondButton.click();
+        await elementUpdated(el);
+
+        expect(el.selected.length).to.equal(1);
+        expect(el.selected).to.deep.equal(['second']);
+        expect(firstButton.selected, 'first button not selected').to.be.false;
+        expect(secondButton.selected, 'second button selected').to.be.true;
+
+        firstButton.click();
+        await elementUpdated(el);
+
+        expect(el.selected.length).to.equal(1);
+        expect(el.selected).to.deep.equal(['first']);
+        expect(firstButton.selected, 'first button not selected').to.be.true;
+        expect(secondButton.selected, 'second button selected').to.be.false;
+
+        const rect = icon.getBoundingClientRect();
+        await sendMouse({
+            steps: [
+                {
+                    type: 'click',
+                    position: [
+                        rect.left + rect.width / 2,
+                        rect.top + rect.height / 2,
+                    ],
+                },
+            ],
+        });
+        icon.click();
```

Comment:
Just FMI, what was happening before on icon click?

Reply:
It would ignore the click. See this early exit in the handler:
```
const target = event.target as ActionButton;
if (typeof target.value === 'undefined') {
    return;
}
```
When the click event happens on the Icon then it won't have a value and it will return. There's likely other issues to handle therein, but I was focusing on the immediate error that we had discovered.